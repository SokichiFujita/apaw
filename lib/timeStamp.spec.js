const timeStamp = require('./timeStamp');

describe('timeStamp', () => {
  let dateSpy;

  beforeAll(() => {
    const mockDate = '2020-03-09T03:33:41.615Z';
    dateSpy = jest.spyOn(global, 'Date').mockImplementation(() => ({
      toISOString() {
        return mockDate;
      },
    }));
  });

  afterAll(() => {
    dateSpy.mockRestore();
  });

  it('should return a "longDate" in a formatted ISO format: yyyymmddThhmmssZ', () => {
    const { longDate } = timeStamp();
    expect(longDate).toMatchSnapshot();
  });

  it('should return a "shortDate" in a formatted ISO format: yyyymmdd', () => {
    const { shortDate } = timeStamp();
    expect(shortDate).toMatchSnapshot();
  });
});