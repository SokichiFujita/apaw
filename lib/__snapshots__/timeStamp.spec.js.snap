// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeStamp should return a "longDate" in a formatted ISO format: yyyymmddThhmmssZ 1`] = `"20200309T033341Z"`;

exports[`timeStamp should return a "shortDate" in a formatted ISO format: yyyymmdd 1`] = `"20200309"`;
